src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestPrimitivesGen.generated.txt

```diff
@@ -0,0 +1,178 @@
+    }
+}
+
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
+{
+    using System;
+    using System.Globalization;
+    using Microsoft.Extensions.Configuration;
+
+    internal static class Helpers
+    {
+        public static void BindCore(IConfiguration configuration, ref Program.MyClass obj)
+        {
+            if (obj is null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            if (configuration["MyString"] is string stringValue1)
+            {
+                obj.MyString = stringValue1;
+            }
+
+            if (configuration["MyInt128"] is string stringValue2)
+            {
+                try
+                {
+                    obj.MyInt128 = Int128.Parse(stringValue2, CultureInfo.InvariantCulture);
+                }
+                catch (Exception exception)
+                {
+                    throw new InvalidOperationException($"Failed to convert configuration value at '{configuration.GetSection("MyInt128").Path}' to type '{typeof(Int128)}'.", exception);
+                }
+            }
+
+            if (configuration["MyInt"] is string stringValue3)
+            {
+                try
+                {
+                    obj.MyInt = int.Parse(stringValue3, NumberStyles.Integer, CultureInfo.InvariantCulture);
+                }
+                catch (Exception exception)
+                {
+                    throw new InvalidOperationException($"Failed to convert configuration value at '{configuration.GetSection("MyInt").Path}' to type '{typeof(int)}'.", exception);
+                }
+            }
+
+            if (configuration["MyUInt128"] is string stringValue4)
+            {
+                try
+                {
+                    obj.MyUInt128 = UInt128.Parse(stringValue4, CultureInfo.InvariantCulture);
+                }
+                catch (Exception exception)
+                {
+                    throw new InvalidOperationException($"Failed to convert configuration value at '{configuration.GetSection("MyUInt128").Path}' to type '{typeof(UInt128)}'.", exception);
+                }
+            }
+
+            if (configuration["MyLong"] is string stringValue5)
+            {
+                try
+                {
+                    obj.MyLong = long.Parse(stringValue5, NumberStyles.Integer, CultureInfo.InvariantCulture);
+                }
+                catch (Exception exception)
+                {
+                    throw new InvalidOperationException($"Failed to convert configuration value at '{configuration.GetSection("MyLong").Path}' to type '{typeof(long)}'.", exception);
+                }
+            }
+
+            if (configuration["MyUri"] is string stringValue6)
+            {
+                Uri.TryCreate(stringValue6, UriKind.RelativeOrAbsolute, out Uri? temp7);
```

Comment:
The UriTypeConverter uses `new Uri` and lets the ctor throw if the value doesn't parse correctly.

[https://github.com/dotnet/runtime/blob/5711ca33f8dd1758e18c09ebdedf7e95eb424e0f/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/UriTypeConverter.cs#L49-L50](https://github.com/dotnet/runtime/blob/5711ca33f8dd1758e18c09ebdedf7e95eb424e0f/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/UriTypeConverter.cs#L49-L50)